sdks/java/extensions/avro/src/test/java/org/apache/beam/sdk/extensions/avro/schemas/AvroSchemaTest.java

```diff
@@ -302,23 +304,8 @@ public String toString() {
   private static final DateTime DATE_TIME =
       new DateTime().withDate(1979, 3, 14).withTime(1, 2, 3, 4);
   private static final LocalDate DATE = new LocalDate(1979, 3, 14);
-  private static final TestAvroNested AVRO_NESTED_SPECIFIC_RECORD = new TestAvroNested(true, 42);
   private static final TestAvro AVRO_SPECIFIC_RECORD =
-      new TestAvro(
-          true,
-          43,
-          44L,
-          (float) 44.1,
-          (double) 44.2,
-          "mystring",
-          ByteBuffer.wrap(BYTE_ARRAY),
-          new fixed4(BYTE_ARRAY),
-          DATE,
-          DATE_TIME,
-          TestEnum.abc,
-          AVRO_NESTED_SPECIFIC_RECORD,
-          ImmutableList.of(AVRO_NESTED_SPECIFIC_RECORD, AVRO_NESTED_SPECIFIC_RECORD),
-          ImmutableMap.of("k1", AVRO_NESTED_SPECIFIC_RECORD, "k2", AVRO_NESTED_SPECIFIC_RECORD));
+      TestAvroFactory.newInstanceForAvroSchemaTest();
```

Comment:
With this change we also stop testings non-empty maps for Avro 1.8.2 :/
How about duplicating the test cases here to test maps both with and without content? The tests using non empty maps could then be excluded from the version tests until fixed for a more recent Avro version. That way tests would also serve as documentation on what is supported and what not (similar to runner validation).

Reply:
Empty maps logic is implemented inside `AvroCoderTest.testSpecificRecordEncoding()` that was failing because of this issue in recent Avro versions.